docs/posts/overview-of-ballerina-0.982.0.md

```diff
@@ -0,0 +1,325 @@
+
+```ballerina
+public type Foo abstract object {
+    public string name;
+    public int id;
+
+    function getName() returns string;
+
+    function getID() returns int;
+};
+```
+
+### Introduction of Record Iteration Support
+
+Records are now an iterable type. With this change, the `foreach` statement and iterable operations can now be used 
+with records. When iterating a record, one can iterate either over the fields (i.e., field name and value pair) or only 
+over the field values.
+
+```ballerina
+type Person record {
+    string name;
+    int age;
+    string address;
+};
+```
+
+The `foreach` statement can be used on an instance of this record as follows:
+
+```ballerina
+    foreach field, value in person {
+        io:println(field + " : " + <string>value);
+    }
+```
+
+Or if iterating only through the field values:
+```ballerina
+    foreach value in person {
+        io:println(value);
+    }
+```
+
+### Changes to the `main` function
+
+The `main` function now needs to marked `public` and the `main` function could now return an int.
+ 
+Any public function in the entry package could now be invoked via `ballerina run`.
+For example, the public function `add` in the `calculator` package could be invoke, with the integer arguments `4` and 
+`5`, via `ballerina run` as follows:
+
+```cmd
+ballerina run calculator:add 4 5
+```
+If no function is specified, the name of the function to run is considered `main`.
+
+The function invoked via `ballerina run` (including the `main` function) would be data-binding and could have zero or 
+more parameters of any supported type, including any number of required/defaultable parameters and a single rest 
+parameter. This function could also return a value.
+
+For example, given the following public function in the package `currency`:
+```ballerina
+function queryChanges(string host, int port = 8080, string… params) returns float {
+
+}
+```
+Invoking this function via `ballerina run` as follows:
+```cmd
+ballerina run --printreturn currency:queryChanges localhost -port=8181 high day
+```
+would result in the following value assignments
+    - `host` ← `"localhost"`
+    - `port` ← `8181`
+    - `params` ← `["high", "day"]`
+
+When the `--printreturn` flag is set, the return value is printed to the standard out stream.
+
+### New Documentation Syntax
+
+Old syntax:
+```ballerina
+documentation {
+    Adds parameter `x` and parameter `y`.
+    P{{x}} one thing to be added
+    P{{y}} another thing to be added
+    R{{}} the sum of them
+}
+function add(int x, int y) returns int { return x + y; }
+```
+
+New syntax:
+```ballerina
+# Adds parameter `x` and parameter `y`.
+# + x - one thing to be added
+# + y - another thing to be added
+# + return - the sum of them
+function add(int x, int y) returns int { return x + y; }
+```
+
+Note: With this change the previous documentation syntax is no longer supported. 
```

Comment:
Minor change:
"Note: Ballerina does not support the old documentation syntax with the new release. "